### PR TITLE
whisper/whisperv6: discard message payload

### PR DESCRIPTION
**Pull request by ucwong**

### CODE REVIEW

File: whisper/whisperv6/whisper.go [hunk 1]
```diff
@@ -696,6 +696,7 @@ func (whisper *Whisper) runMessageLoop(p *Peer, rw p2p.MsgReadWriter) error {
 		}
 		if packet.Size > whisper.MaxMessageSize() {
 			log.Warn("oversized message received", "peer", p.peer.ID())
+			packet.Discard()
```
Comment (gballet) [inline whisper/whisperv6/whisper.go:699]: `defer packet.Discard()` ?

Comment (ucwong) [reply whisper/whisperv6/whisper.go:699]: It will return immediately, I think ```packet.Discard()``` is enough here @gballet

**Comment by ucwong** [comment]
Discard will be only called once before the loop ended in this pr, and make sure the msg payload is already alloc.
I think it will be a little better to free memory immediately than gc invoke.

**Comment by fjl** [comment]
I think this change is not really needed. The `Discard` doesn't really have anything to do with memory, it just ensures the message stream has been read up to the beginning of the next message.

**Pull request closed without merging**